.github/workflows/helm-test.yml

```diff
@@ -112,21 +112,17 @@ jobs:
                     ct install --charts charts/puris --helm-extra-set-args "--set backend.image.repository=${{ env.REGISTRY }}/${{ env.APP_BACKEND_NAME}} --set backend.image.tag=${{ env.TAG }} --set frontend.image.repository=${{ env.REGISTRY }}/${{ env.APP_FRONTEND_NAME }} --set frontend.image.tag=${{ env.TAG }}"
                 if: github.event_name != 'pull_request' || steps.list-changed.outputs.changed == 'true'
 
-            -   name: Generate random password (This password is only a placeholder for the next step and will not used).
-                id: generate-password
-                run: |
-                    echo "PASSWORD=PLACEHOLDER_PW" >> $GITHUB_ENV
-
             # Upgrade the released chart version with the locally available chart
             # default value for event_name != workflow_dispatch
             # Setting passwords during upgrade:
-            # - The postgresPassword/password will be set via postgres-init.yaml.
-            # - The dtr.idp.clients.edc.puris.secret is new and needs to be set during upgrade as not set in previous version
+            # - set existing secret to value currently defaulted
+            # - insert password to postgres directly due to empty default for reuse during upgrade
+            # https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases
             -   name: Run helm upgrade
                 run: |
                     helm repo add bitnami https://charts.bitnami.com/bitnami
                     helm repo add tractusx-dev https://eclipse-tractusx.github.io/charts/dev
-                    helm install puris tractusx-dev/puris --version ${{ github.event.inputs.upgrade_from || '1.0.0' }}
+                    helm install puris tractusx-dev/puris --version ${{ github.event.inputs.upgrade_from || '1.0.0' }} --set postgresql.auth.existingSecret="secret-puris-postgres-init"
                     helm dependency update charts/puris
-                    helm upgrade puris charts/puris --set backend.puris.datasource.password=$PASSWORD --set backend.puris.dtr.idp.clients.puris.secret=$PASSWORD
+                    helm upgrade puris charts/puris --set global.postgresql.auth.postgres=$(kubectl get secret -n default secret-puris-postgres-init -o jsonpath="{.data.password}" | base64 --decode)
```

Comment:
are you sure that you still nee that set flag?
I'd assume that you'd not need it anymore because you did the [`helm install` with the secret name](https://github.com/eclipse-tractusx/puris/pull/389/files#diff-33f5fdd3507841f6bb8f87595ecc5669a0397021de950482ca4e8b68c25cfdb6R125) to which the chart defaults to at upgrade

Reply:
actually yes. Tried it first without but then the deployment failed and stated to this [url](https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases) telling me to set a global config so that postgres does not create a new password.